ruddock/modules/users/routes.py

```diff
@@ -26,17 +27,64 @@ def show_memberlist(search_type):
       memberlist=memberlist,
       search_type=search_type)
 
-@blueprint.route('/view/<username>')
+@blueprint.route('/view/user_id=<user_id>')
 @login_required()
-def view_profile(username):
-  """Procedure to show a user's profile and membership details."""
-  user_info = helpers.get_user_info(username)
-  offices = helpers.get_office_info(username)
+def view_member(user_id):
+  """Procedure to show a member's profile and membership details if they have
+  no user account on the website."""
+  if user_id is not None:
```

Comment:
both get_member_info and get_office_info are safe to pass anything into, including None

so you can just drop this if-statement, and rely on "if member_info is not None" to do the error handling for you